File: 7_Function_Overloading.cpp
// Function overloading in C++ allows you to define multiple functions with the same name but with different parameters.

#include<bits/stdc++.h>
using namespace std;

// Function to calculate the area of a square
int area(int side) {
    return side * side;
}

// Function to calculate the area of a rectangle
int area(int length, int breadth) {
    return length * breadth;
}

// Function to calculate the area of a circle
float area(float radius) {
    return 3.14f * radius * radius;
}

int main() {
    int squareSide = 5;
    int rectangleLength = 4;
    int rectangleBreadth = 6;
    float circleRadius = 3.5;

    // Call area function for different shapes
    std::cout << "Area of square: " << area(squareSide) << std::endl;
    std::cout << "Area of rectangle: " << area(rectangleLength, rectangleBreadth) << std::endl;
    std::cout << "Area of circle: " << area(circleRadius) << std::endl;

    return 0;
}

File: 7_Function_Overloading.py
# In Python, function overloading is not directly supported as in C++

import math

# Function to calculate the area of a square
def area(side):
    return side * side

# Function to calculate the area of a rectangle
def area_rectangle(length, breadth):
    return length * breadth

# Function to calculate the area of a circle
def area_circle(radius):
    return math.pi * radius * radius

def main():
    square_side = 5
    rectangle_length = 4
    rectangle_breadth = 6
    circle_radius = 3.5

    # Call area functions for different shapes
    print("Area of square:", area(square_side))
    print("Area of rectangle:", area_rectangle(rectangle_length, rectangle_breadth))
    print("Area of circle:", area_circle(circle_radius))

if __name__ == "__main__":
    main()
